grammar.js

```diff
@@ -1,489 +1,653 @@
-				field('operand', $._expression),
-		)))),
-
-		_primary_expression: $ => choice(
-			$._operand,
-			$.index_expression,
-			$.selector_expression,
-			$.call_expression,
-		),
-
-		_operand: $ => choice(
-			$.identifier,
-			$._literal,
-			seq('(', $._expression, ')'),
-		),
-
-		_call_operand: $ => choice(
-			$.builtin,
-			$.qualified_identifier,
-		),
-
-		index_expression: $ => seq($._primary_expression, $.index),
-
-		selector_expression: $ =>	seq($._primary_expression, $.selector),
-
-		call_expression: $ => seq($._call_operand, $.arguments),
-
-		selector: $ => seq($.dot, choice($.identifier, $.simple_string_lit)),
-
-		index: $ => seq(token.immediate('['), $._expression, ']'),
-
-		arguments: $ => seq(
-			'(', $.argument, repeat(seq(',', $.argument)), ')'
-		),
-		argument: $ => $._expression,
-
-		_literal: $ => choice(
-			$.struct_lit,
-			$.list_lit,
-			$._string_lit,
-			$.string_type,
-			$.bytes_type,
-			$.number_type,
-			$.int_lit,
-			$.int_type,
-			$.uint_type,
-			$.float_lit,
-			$.float_type,
-			$.null,
-			$.true,
-			$.false,
-			$.bool_type,
-			$.top_lit,
-			$.bottom_lit,
-		),
-
-		comment: $ => token(choice(
-			seq('//', /.*/),
-		))
-
-	}
+  name: 'cue',
+
+  extras: $ => [
+    $.comment,
+    /\s/,
+  ],
+
+  externals: $ => [
+    $._multi_str_content,
+    $._multi_bytes_content,
+    $._raw_str_content,
+    $._raw_bytes_content,
+    $._multi_raw_str_content,
+    $._multi_raw_bytes_content,
+  ],
+
+  inline: $ => [
+    $.keyword_identifier,
+  ],
+
+  supertypes: $ => [
+    $.expression,
+    $.primary_expression,
+  ],
+
+  conflicts: $ => [
+    [$._embedding, $._label_alias_expr],
+  ],
+
+  word: $ => $.identifier,
+
+  rules: {
+    source_file: $ => seq(
+      optional($.attribute),
+      optional($.package_clause),
+      optional(repeat($.import_declaration)),
+      optionalCommaSep($._declaration),
+    ),
+
+    _package_identifier: $ => alias($.identifier, $.package_identifier),
+
+    package_clause: $ => seq(
+      'package',
+      $._package_identifier,
+    ),
+
+    import_declaration: $ => seq(
+      'import',
+      choice(
+        $.import_spec,
+        $.import_spec_list,
+      ),
+    ),
+
+    import_spec: $ => seq(
+      optional(field('name', choice(
+        '.',
+        '_',
+        $._package_identifier,
+      ))),
+      field('path', $.string),
+    ),
+
+    import_spec_list: $ => seq(
+      '(',
+      optionalCommaSep($.import_spec),
+      ')',
+    ),
+
+    identifier: _ => token(choice(
+      seq(
+        optional(choice('_#', '#', '_')),
+        choice(/\p{L}/, '$', '_'),
+        repeat(choice(/\p{L}/, '$', '_', /[0-9]/)),
+      ),
+      choice('_#', '#'),
+    )),
+    keyword_identifier: $ => prec(-3, alias(
+      choice(
+        'let',
+        'if',
+        'for',
+        ...primitive_types,
+      ),
+      $.identifier,
+    )),
+
+    attribute: $ => seq(
+      '@',
+      $._attr_token,
```

Comment:
```suggestion
      $.identifier,
```

Reply:
Hmm why? You had such a token beforehand.. but I made some custom changes to it to improve parsing and syntax highlighting for attributes instead of parsing it all as a single token

Reply:
I believe you want to parse the attributes parameters `@attr(params...)` but the name of the attribute should be constrained like other identifiers:

```
attribute       = "@" identifier "(" attr_tokens ")" .
attr_tokens     = { attr_token |
                    "(" attr_tokens ")" |
                    "[" attr_tokens "]" |
                    "{" attr_tokens "}" } .
attr_token      = /* any token except '(', ')', '[', ']', '{', or '}' */
```

Reply:
Oh I see what you mean, thanks! Sorry about that